tools/generator/make_rules.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python3
+
+import jinja2
+import os
+import sys
+import yaml
+from pathlib import Path
+from shutil import copyfile
+
+
+def cxx_type(value: str) -> str:
+    if value == "string":
+        return "std::string"
+    if value.endswith(" array"):
+        return "std::vector<{}>".format(cxx_type(value[:-6]))
+    return value
+
+def cxx_args(value) -> str:
+    return ", ".join("{} {}".format(cxx_type(i[1]), i[0]) for i in value)
+
+def snake_to_camel_case(value: str) -> str:
+    """Convert a snake case identifier to a upper camel case one"""
+    return "".join(i.capitalize() for i in value.split("_"))
```

Comment:
I think it would be better to have a |camel_case and a |snake_case filter that are both idempotent and accept both cases as their input. It sounds doable.

Reply:
I don’t really see the point.
The functions are written in the yaml in snake_case. And in some places, it will be converted in camel case. That is why I named the function `snake_to_camel_case` instead of the previous `camel_case`.
So the name is a bit long, but it is the only case transformation we’ll be needing.